test/utils/mochaw.js

```diff
@@ -1,12 +1,26 @@
 import mocha from 'mocha';
-const {By} = require('selenium-webdriver');
+const {By, until} = require('selenium-webdriver');
 const expect = require('chai').expect
 
 const $driver = driver => selector =>
   driver.findElement(By.css(selector))
 
+const waitAndFindElement = driver => selector => {
+  const locator = By.css(selector)
+  return driver.findElement(async () => {
+    await driver.wait(until.elementLocated(locator))
+    return driver.findElement(locator)
+  })
+}
+
+export const click = (driver) => async (element) => {
+  await driver.wait(until.elementIsVisible(element))
+  await driver.wait(until.elementIsEnabled(element))
+  return element.click()
+}
```

Comment:
Does this click replace the native click method provided by Selenium?

Reply:
It doesn't. I think it's a good idea to use this function in the following cases:
- when the button might go from a disabled to an enabled state
- when the button is displayed asynchronously
- when you want to click on an element on a screen that takes some time to load
We could replace the native Selenium click method eventually, but for now I only replaced it for the elements that were broken.

Reply:
This click function is actually renamed as `clickWhenClickable` in the page objects and you use it as in the following example `this.clickWhenClickable(this.continueButton())`

Reply:
Thank you for explaining.